File: ios/FOSinOC/FOSinOC/Login/TutorialRootViewController.m
//
//  TutorialRootViewController.m
//  FOSinOC
//
//  Created by MoonSlides on 16/4/7.
//  Copyright © 2016年 李龑. All rights reserved.
//

#import "TutorialRootViewController.h"


@interface TutorialRootViewController ()


@property (strong, nonatomic) NSArray *arrayPageTitle;
@property (strong, nonatomic) NSArray *arrayPageText;
@property (strong, nonatomic) NSArray *arrayImage;




@end

@implementation TutorialRootViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    NSString *introText1 = @"Meals arrive on-demand or you can schedule delivery up to a week in advance for beyond-easy meal-planning.";
    NSString *introText2 = @"Our friendly delivery folk arrive with your food, still chilled so it stays fresh. We’ll bring it right to your front door. Or your desk. Or your friend's place.";
    NSString *introText3 = @"Mealtimes are more than food. They’re time to relax. Sometimes that means recharging all by yourself, other times it’s about connecting with friends and family. However you eat dinner, we’re here to make it as easy as it is delicious.";
    
    self.arrayPageTitle = @[@"DINNER MADE EASY", @"DELIVERED ANYWHERE", @"DINNERS YOU WANT"];
    self.arrayImage = @[@"introductionPic1.jpg", @"introductionPic2.jpg", @"introductionPic3.jpg"];
    self.arrayPageText = @[introText1, introText2, introText3];
    
    self.pageViewController = [[self storyboard] instantiateViewControllerWithIdentifier:@"PageViewController"];
    self.pageViewController.dataSource = self;
    
    TutorialPageViewController *startingViewController = [self viewControllerAtIndex:0];
    
    NSArray *viewControllers = [NSArray arrayWithObject:startingViewController];
   
    [self.pageViewController setViewControllers:viewControllers direction:UIPageViewControllerNavigationDirectionForward animated:NO completion:nil];
    
    self.pageViewController.view.frame = CGRectMake(0, 0, self.view.frame.size.width, self.view.frame.size.height);
    [self addChildViewController:self.pageViewController];
    [self.view addSubview:self.pageViewController.view];
    [self.pageViewController didMoveToParentViewController:self];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


-(TutorialPageViewController *) viewControllerAtIndex:(NSUInteger) index {
    if((self.arrayPageTitle.count == 0) || (index >= self.arrayPageTitle.count)) {
        return nil;
    }
    TutorialPageViewController *pageContentViewController = [[self storyboard] instantiateViewControllerWithIdentifier:@"PageContentViewController"];
   
    pageContentViewController.imageFile = self.arrayImage[index];
    pageContentViewController.textTitle = self.arrayPageTitle[index];
    pageContentViewController.textContent = self.arrayPageText[index];
    pageContentViewController.pageIndex = index;
    return pageContentViewController;
}
-(BOOL)prefersStatusBarHidden{
    return YES;
}

-(UIViewController *)pageViewController:(UIPageViewController *)pageViewController viewControllerBeforeViewController:(UIViewController *)viewController{
    NSUInteger index = [(TutorialPageViewController *)viewController pageIndex];
    if (index == NSNotFound || index == 0){
        return nil;
    }
    index --;
    return [self viewControllerAtIndex:index];
    
}

-(UIViewController *) pageViewController:(UIPageViewController *)pageViewController viewControllerAfterViewController:(UIViewController *)viewController{
    NSUInteger index = [(TutorialPageViewController *)viewController pageIndex];
    if (index == NSNotFound){
        return nil;
    }
    index ++;
    if (index == _arrayPageTitle.count){
        return nil;
    }
    return [self viewControllerAtIndex:index];
}

-(NSInteger)presentationCountForPageViewController:(UIPageViewController *)pageViewController{
    return self.arrayPageText.count;
}

-(NSInteger)presentationIndexForPageViewController:(UIPageViewController *)pageViewController{
    return 0;
}
@end

File: ios/FoodOrderingSystem/FoodOrderingSystem/IntroAndLogin/TutorialRootViewController.swift
//
//  ViewController.swift
//  FoodOrderingSystem
//
//  Created by MoonSlides on 16/1/16.
//  Copyright © 2016年 李龑. All rights reserved.
//

import UIKit

class TutorialRootViewController: UIViewController, UIPageViewControllerDataSource, AccountDelegate {

    var pageViewController: UIPageViewController?
    var arrayPageTitles: NSArray?
    var arrayPageTexts: NSArray?
    var arrayImages: NSArray?
    let introText1 = "Meals arrive on-demand or you can schedule delivery up to a week in advance for beyond-easy meal-planning."
    let introText2 = "Our friendly delivery folk arrive with your food, still chilled so it stays fresh. We’ll bring it right to your front door. Or your desk. Or your friend's place."
    let introText3 = "Mealtimes are more than food. They’re time to relax. Sometimes that means recharging all by yourself, other times it’s about connecting with friends and family. However you eat dinner, we’re here to make it as easy as it is delicious."
    
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        if let token = NSUserDefaults.standardUserDefaults().stringForKey("token") {
            let account = Account.sharedManager
            account.token = token
            account.delegate = self
            account.checkLoginStatus()
        }
        
        arrayPageTitles = ["DINNER MADE EASY", "DELIVERED ANYWHERE", "DINNERS YOU WANT"]
        arrayImages = ["introductionPic1.jpg", "introductionPic2.jpg", "introductionPic3.jpg"]
        arrayPageTexts = [introText1, introText2, introText3]
        self.pageViewController = self.storyboard?.instantiateViewControllerWithIdentifier("PageViewController") as? UIPageViewController
        self.pageViewController?.dataSource = self
        let startingViewController = self.viewControllerAtIndex(0)
        let viewControllers: NSArray = [startingViewController!]
        self.pageViewController?.setViewControllers(viewControllers as? [UIViewController], direction: .Forward, animated: false, completion: nil)
        
        self.pageViewController!.view.frame = CGRectMake(0, 0, self.view.frame.width, self.view.frame.height)
        self.addChildViewController(pageViewController!)
        self.view.addSubview(pageViewController!.view)
        self.pageViewController?.didMoveToParentViewController(self)
        //performSegueWithIdentifier("login", sender: nil)
        
    }
    
    override func viewDidAppear(animated: Bool) {
        super.viewDidAppear(animated)
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    override func prefersStatusBarHidden() -> Bool {
        return true
    }
    
    func viewControllerAtIndex(index: UInt) -> TutorialPage? {
        if ((self.arrayPageTitles!.count == 0) || (Int(index) >= self.arrayPageTitles!.count)) {
            return nil
        }
        let pageContentViewController = self.storyboard?.instantiateViewControllerWithIdentifier("PageContentViewController") as! TutorialPage
        pageContentViewController.imageFile = self.arrayImages![Int(index)] as! String
        pageContentViewController.textTitle = self.arrayPageTitles![Int(index)] as! String
        pageContentViewController.textContents = self.arrayPageTexts![Int(index)] as! String
        
        pageContentViewController.pageIndex = index
        return pageContentViewController
    }


    
    func pageViewController(pageViewController: UIPageViewController, viewControllerBeforeViewController viewController: UIViewController) -> UIViewController? {
        var index = (viewController as! TutorialPage).pageIndex
        if (index == 0 || Int(index) == NSNotFound) {
            return nil;
        }
        index--
        return self.viewControllerAtIndex(index)
    }
    
    func pageViewController(pageViewController: UIPageViewController, viewControllerAfterViewController viewController: UIViewController) -> UIViewController? {
        var index = (viewController as! TutorialPage).pageIndex
        
        if Int(index) == NSNotFound {
            return nil
        }
        
        index++
        
        if Int(index) == arrayPageTitles!.count {
            return nil
        }
        
        return self.viewControllerAtIndex(index)
    }
    
    func presentationCountForPageViewController(pageViewController: UIPageViewController) -> Int {
        return self.arrayPageTitles!.count
    }
    
    func presentationIndexForPageViewController(pageViewController: UIPageViewController) -> Int {
        return 0
    }
}

extension TutorialRootViewController {


    func finishLogin(result: NSDictionary, account: Account) {
        if let success = result.objectForKey("success") {
            if success as! NSObject == true {
//                let storyboard = UIStoryboard(name: "SlideMenu", bundle: nil)
//                let mainMenu = storyboard.instantiateViewControllerWithIdentifier("slideMenu")
//                presentViewController(mainMenu, animated: true, completion: nil)
                
            }
        }
    }
}
